Platform/CommonBoardPkg/Library/TccLib/TccLib.c

```diff
@@ -0,0 +1,639 @@
+/** @file
+
+  Copyright (c) 2021 Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+**/
+
+#include <PiPei.h>
+#include <Library/BaseLib.h>
+#include <Library/DebugLib.h>
+#include <Library/PciLib.h>
+#include <Library/IoLib.h>
+#include <Library/MemoryAllocationLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/BootloaderCoreLib.h>
+#include <Library/HobLib.h>
+#include <Library/BootloaderCommonLib.h>
+#include <Library/ContainerLib.h>
+#include <Library/TccLib.h>
+#include <Library/TccPlatformLib.h>
+#include "TccInfo.h"
+
+#define SIGNATURE_IPFW    SIGNATURE_32 ('I', 'P', 'F', 'W')
+#define SIGNATURE_TCCT    SIGNATURE_32 ('T', 'C', 'C', 'T')
+#define SIGNATURE_TCCB    SIGNATURE_32 ('T', 'C', 'C', 'B')
+#define SIGNATURE_TCCC    SIGNATURE_32 ('T', 'C', 'C', 'C')
+#define SIGNATURE_TCCM    SIGNATURE_32 ('T', 'C', 'C', 'M')
+
+TCC_INFO_HOB    *mTccInfoHob    = NULL;
+BIOS_SETTINGS   *PolicyConfig;
+TCC_BINARIES_BASE *TccBinaries;
+UINT8 mTccRtd3Support;
+UINT8 mTccLowPowerS0Idle;
+
+/**
+  Update FSP-M UPD config data for TCC tuning, load TCC buffer & tuning data binaries from TempRam
+**/
+
+EFI_STATUS
+EFIAPI
+TccModePreMemConfig (
+  OUT BIOS_SETTINGS*     PreMemPolicyConfig,
+  OUT TCC_BINARIES_BASE* PreMemTccBinaries
+)
+{
+  TCC_STREAM_CONFIGURATION              *PreMemStreamConfig;
+  EFI_STATUS                            Status;
+  VOID                                  *TccPreMemBufferBase;
+  UINT32                                TccPreMemBufferSize;
+  VOID                                  *TccPreMemTuningBase;
+  UINT32                                TccPreMemTuningSize;
+
+  DEBUG ((DEBUG_INFO, "TccModePreMemConfig () - Start\n"));
+
+  // Load TCC tuning data from container
+  Status = LoadComponent (SIGNATURE_IPFW, SIGNATURE_TCCT, &TccPreMemTuningBase, &TccPreMemTuningSize);
+  if (EFI_ERROR (Status) || AsciiStrCmp ((CHAR8*) TccPreMemTuningBase, "dummy\0") == 0 ||
+      TccPreMemTuningSize < sizeof (TCC_STREAM_CONFIGURATION)) {
+    // Set default tuning values for TCC mode if TCC Tuning binary is not found
+    DEBUG ((DEBUG_INFO, "TCC Tuning data not found! Using default TCC tuning policies.\n"));
+    PreMemPolicyConfig               = NULL;
+    TccPreMemTuningBase        = NULL;
+    TccPreMemTuningSize        = 0;
+
+  } else {
+    // Consume TCC tuning binary settings
+    PreMemStreamConfig = (TCC_STREAM_CONFIGURATION *) TccPreMemTuningBase;
+    PreMemPolicyConfig = (BIOS_SETTINGS *) &PreMemStreamConfig->Info.Format1.BiosSettings;
+  }
+
+  // Load Tcc buffers data from container
+  Status = LoadComponent (SIGNATURE_IPFW, SIGNATURE_TCCB, &TccPreMemBufferBase, &TccPreMemBufferSize);
+  if (EFI_ERROR (Status) || AsciiStrCmp ((CHAR8*) TccPreMemBufferBase, "dummy\0") == 0) {

```

Comment:
If we are not checking in Tcc binaries with keyword dummy, then this dummy check may not be needed. Initially, I believe we had a dummy string originally inside the binary to differentiate the actual data. Checking for BufferBase NULL and checking for the struct size should be good enough I feel. Please verify.